src/parser/mod.rs

```diff
@@ -10511,13 +10517,79 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let with_fill = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keywords(&[Keyword::WITH, Keyword::FILL])
+        {
+            Some(self.parse_with_fill()?)
+        } else {
+            None
+        };
+
         Ok(OrderByExpr {
             expr,
             asc,
             nulls_first,
+            with_fill,
         })
     }
 
+    // Parse a WITH FILL clause (ClickHouse dialect)
+    // that follow the WITH FILL keywords in a ORDER BY clause
+    pub fn parse_with_fill(&mut self) -> Result<WithFill, ParserError> {
+        let from = if self.parse_keyword(Keyword::FROM) {
+            Some(self.parse_expr()?)
+        } else {
+            None
+        };
+
+        let to = if self.parse_keyword(Keyword::TO) {
+            Some(self.parse_expr()?)
+        } else {
+            None
+        };
+
+        let step = if self.parse_keyword(Keyword::STEP) {
+            Some(self.parse_expr()?)
+        } else {
+            None
+        };
+
+        Ok(WithFill { from, to, step })
+    }
+
+    // Parse a set of comma seperated INTERPOLATE expressions (ClickHouse dialect)
+    // that follow the INTERPOLATE keyword in an ORDER BY clause with the WITH FILL modifier
+    pub fn parse_interpolations(&mut self) -> Result<Option<Interpolate>, ParserError> {
+        if dialect_of!(self is ClickHouseDialect | GenericDialect)
```

Comment:
We can move this dialect check outside the function and to the caller, and have the function only care about the parsing logic.

one minor, we can also avoid nesting the function body by returning early
```rust
if !self.parse_keyword(Keyword::INTERPOLATE) {
     return Ok(None)
}

if self.consume_token(...)
Ok(Some(...))
```

Reply:
Done in [000e08d](https://github.com/sqlparser-rs/sqlparser-rs/pull/1330/commits/000e08d3f0849b93b70f056614b51539f3adebfe). Thanks!